modules/overviews/db/migrate/20190826083604_my_project_page_to_grid.rb

```diff
@@ -0,0 +1,255 @@
+class MyProjectPageToGrid < ActiveRecord::Migration[5.2]
+  class MyPageEntry < ActiveRecord::Base
+    self.table_name = 'my_projects_overviews'
+
+    serialize :top
+    serialize :left
+    serialize :right
+
+    belongs_to :project
+  end
+
+  def up
+    return unless applicable?
+
+    recreate_my_page_entries
+    remove_my_page_table
+    add_permission
+  end
+
+  def down
+    # down migration will loose data
+    Grids::Overview.destroy_all
+
+    create_table :my_projects_overviews, id: :integer do |t|
+      t.integer "project_id", default: 0, null: false
+      t.text "left", null: false
+      t.text "right", null: false
+      t.text "top", null: false
+      t.text "hidden", null: false
+      t.datetime "created_on", null: false
+    end
+  end
+
+  def recreate_my_page_entries
+    migratable_entries.in_batches(of: 200).each do |entries|
+      entries.each do |entry|
+        grid = create_grid(entry)
+        move_attachments(entry, grid)
+      end
+    end
+  end
+
+  def remove_my_page_table
+    drop_table :my_projects_overviews
+  end
+
+  def add_permission
+    Role
+      .includes(:role_permissions)
+      .where(role_permissions: { permission: 'edit_project' })
+      .each do |role|
+      role.add_permission!(:manage_overview)
+    end
+  end
+
+  def create_grid(entry)
+    grid = Grids::Overview.new project: entry.project, column_count: 2, created_at: entry.created_on
+
+    %i[top left right].each do |area|
+      entry.send(area).each do |widget|
+        build_widget(grid, widget, area)
+      end
+    end
+
+    grid.row_count = grid.widgets.map(&:end_row).max || 1
+
+    grid.save!
+    grid
+  end
+
+  def move_attachments(entry, grid)
+    Attachment
+      .where(container_type: "MyProjectsOverview", container_id: entry.id)
+      .update_all(container_type: 'Grids::Grid', container_id: grid.id)
+  end
+
+  def build_widget(grid, widget_config, position)
+    method = case widget_config
+             when Array
+               :build_custom_text_widget
+             when 'project_details'
+               :build_project_details_widget
+             when 'work_packages_assigned_to_me',
+                  'work_packages_reported_by_me',
+                  'work_packages_responsible_for',
+                  'work_packages_watched'
+               :build_wp_table_widget
+             when 'project_description',
+                  'work_package_tracking',
+                  'members',
+                  'news_latest',
+                  'calendar',
+                  'spent_time',
+                  'subprojects'
+               :build_default_widget
```

Comment:
This fails for the `QA edge` dump due to the `widget_config` being 'documents'. Adding this in the last when case works, but maybe this is a case for adding a default case instead?